src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -29,43 +33,132 @@ public static IServiceCollection AddBlobDataStore(this IServiceCollection servic
                 return services;
             }
 
-            services.Add(provider =>
+            services.AddSingleton(
+                provider =>
                 {
                     var config = new BlobDataStoreConfiguration();
-                    provider.GetService<IConfiguration>().GetSection(BlobStoreConfigurationSectionName).Bind(config);
-                    configureAction?.Invoke(config);
+                    provider
+                        .GetService<IConfiguration>()
+                        .GetSection(BlobDataStoreConfiguration.SectionName)
+                        .Bind(config);
 
-                    if (string.IsNullOrEmpty(config.ConnectionString) && config.AuthenticationType == BlobDataStoreAuthenticationType.ConnectionString)
-                    {
-                        config.ConnectionString = BlobLocalEmulator.ConnectionString;
-                    }
+                    configureAction?.Invoke(config);
+                    DefaultBlobDataStoreConfiguration.Instance.Configure(config);
```

Comment:
seems we could call `services.ConfigureOptions<DefaultBlobDataStoreConfiguration>()` outside this singleton call, then options system can automatically configure it.

Reply:
Yeah I've refactored this